### ISSUE TITLE
Add support to the Helix editor

### ISSUE THREAD

**Issue by treeman** [treeman/tree-sitter-djot#10]
1. Add it to `languages.toml`: https://github.com/helix-editor/helix/blob/master/languages.toml
2. Insert the queries here: https://github.com/helix-editor/helix/tree/master/runtime/queries

**Comment by StratusFearMe21** [comment]
That would be amazing! You might have to have a separate directory for Helix queries though, since it seems that some of the `nvim-treesitter` queries aren't compatible with Helix, ex. `@markup.strong` is `@markup.bold` in Helix

**Comment by treeman** [comment]
Yes, all the highlights needs to be reworked for the specific editor. The highlight files in this repo have been made with Neovim in mind because that's what I use, but should be seen as a starting point for porting it to other editors. 

And even for Neovim they're a little bit different, as [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) manages `highlights.scm` themselves, which is what everyone using Neovim will probably use (unless they create their own query file).

**Comment by mmibbetson** [comment]
I've been working on this since coming across [this issue](https://github.com/helix-editor/helix/issues/12264#issuecomment-2589482423) in the Helix repo the other day.

As I've said there, work is complete on the highlights and injections --- I'm just finishing up the languages.toml portion before I prepare a PR there.

I've noted that you're working on a pretty big rewrite, I'm happy to adapt the queries once djot and djot-inline are separated, but I figure it would still be nice for us to have current djot support in the meantime in Helix.

**Comment by mmibbetson** [comment]
If you're curious about what the highlights are looking like in Helix, I've provided some screenshots in [this comment on the PR](https://github.com/helix-editor/helix/pull/12562#issuecomment-2598673698).

**Comment by treeman** [comment]
I managed to rework the big rewrite to not be as breaking, although a few breaking changes were made. See https://github.com/treeman/tree-sitter-djot/pull/56 for (hopefully) a complete list of changes.

**Comment by mmibbetson** [comment]
Thanks for your hard work on the rewrite! It happened to coincide quite closely with Helix merging a change to have the same precedence as Neovim in its queries. I've inverted the order and everything should now be in a state that can be merged and will be easy to maintain over time since it's closer to the queries you maintain in this repo as well 😄

**Comment by mmibbetson** [comment]
@treeman helix-editor/helix#12562 has been merged! I think this issue can be closed now.

**Comment by treeman** [comment]
Well done!
